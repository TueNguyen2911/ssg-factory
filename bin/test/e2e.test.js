const runSSG = require("./runSSG"); 

describe("End to End testing", () => {
    test("should print usage and options when --help option is specified", async () => {
        const { stderr, stdout, exitCode} = await runSSG("--help");

        expect(stdout).toMatchSnapshot();
        expect(stderr).toEqual("");    
    });
    // test("should print inuput path and file converting result when --input is specified", async () => {
    //     const { stderr, stdout, exitCode } = await runSSG("--input", "./bin/test/samples");

    //     expect(stdout).toMatchSnapshot();
    //     expect(stderr).toEqual("");    
    // });
    test("should print the version of the program when --version is specified", async () => {
        const { stderr, stdout, exitCode } = await runSSG("--version");

        expect(stdout).toMatchSnapshot();
        expect(stderr).toEqual("");    
    });
    // test("should print the input/output path and result when --config is specified", async () => {
    //     const { stderr, stdout, exitCode } = await runSSG("--config", "./bin/test/config.json");

    //     expect(stdout).toMatchSnapshot();
    //     expect(stderr).toEqual("");    
    // });
})