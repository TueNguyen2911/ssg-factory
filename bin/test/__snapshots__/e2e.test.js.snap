// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`End to End testing should print the version of the program when --version is specified 1`] = `"ssg-factory v1.0.6"`;

exports[`End to End testing should print usage and options when --help option is specified 1`] = `
"Usage: ssg-factory [options]

Options:
  -v, --version                    output the version number
  -o, --output <path>              specify a path for .html files output
  -l, --lang <language code>       adding a language to HTML document
  -i, --input <file path>          (required) transform .txt or .md files into
                                   .html files
  -c, --config <config file path>  use stored commands in config file
  -h, --help                       display help for command"
`;
